build-tools/automation/azure-pipelines.yml

```diff
@@ -0,0 +1,57 @@
+# Xamarin.Android Pipeline
+
+trigger:
+  - master
+  - d16*
+
+stages:
+- stage: Build
+  jobs:
+  - job: Mac
+    pool: Hosted Mac Internal
+    timeoutInMinutes: 240
+    cancelTimeoutInMinutes: 5
+    variables:
+      configuration: Release
+      msbuildArgs: /p:AutoProvision=True /p:AutoProvisionUsesSudo=True /p:IgnoreMaxMonoVersion=False
+      JavaSdkDirectory: /Library/Java/JavaVirtualMachines/jdk1.8.0_144.jdk/Contents/Home/
+      JAVA_HOME: $(JavaSdkDirectory)
+    steps:
+    - script: make prepare-external-git-dependencies CONFIGURATION=$(configuration)
```

Comment:
The use of `make` rather implies that this is macOS/Linux only.  Is it plausible to instead use MSBuild for both macOS & Windows support?

Reply:
Considering that a complete commercial build currently requires a macOS system and that this job is assigned to a mac environment, I think it's alright to continue to rely on unix tools for this build stage. I am hoping to extend this pipeline in the near future to include a stage which will replace our `Xamarin.Android (Win)` azp check which can consume artifacts from this initial macOS stage, and then run the build and test steps which execute on Windows similar to how our builds are structured today. This will allow us to have Windows and macOS checks running and passing against each PR/commit that don't rely on a bundle created elsewhere.